ext/intl/locale/locale.stub.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+class Locale
+{
+    /** @return string */
+    public static function getDefault() {}
+
+    /** @return bool */
+    public static function setDefault(string $locale) {}
+
+    /** @return string|null */
+    public static function getPrimaryLanguage(string $locale) {}
+
+    /** @return string|null */
+    public static function getScript(string $locale) {}
+
+    /** @return string|null */
+    public static function getRegion(string $locale) {}
+
+    /** @return array|false|null */
+    public static function getKeywords(string $locale) {}
+
+    /** @return string|false */
+    public static function getDisplayScript(string $locale, ?string $in_locale = null) {}
+
+    /** @return string|false */
+    public static function getDisplayRegion(string $locale, ?string $in_locale = null) {}
+
+    /** @return string|false */
+    public static function getDisplayName(string $locale, ?string $in_locale = null) {}
+
+    /** @return string|false */
+    public static function getDisplayLanguage(string $locale, ?string $in_locale = null) {}
+
+    /** @return string|false */
+    public static function getDisplayVariant(string $locale, ?string $in_locale = null) {}
+
+    /** @return string|false */
+    public static function composeLocale(array $subtags) {}
+
+    /** @return array */
+    public static function parseLocale(string $locale) {}
```

Comment:
Hrmpf. I just realized that `INTL_CHECK_LOCALE_LEN()` can add a null default value to pretty much everything here.

Reply:
@cmb69 I'm a bit unsure about what to do here. It looks like by default ext/intl simply suppresses all warnings and exceptions, unless you enable intl.error_level or intl.use_exceptions. I feel like certain kinds of errors would do well with always being exceptions (such as the locale length check).

Reply:
@nikic, that's similar to `SQLite3::enableExceptions()`, by the way. I agree that some cases should always throw, regardless of whether the user requested warnings only. `INTL_CHECK_LOCALE_LEN()` certainly looks like such a case.

Reply:
Ah, somehow I forgot about this function. :( (Maybe I was thinking that it always throws?) Anyway, should I add the `null` return values for now wherever needed or should this error be promoted to an exception as Nikita said?

Reply:
I pushed a fix which adds the missing `null` return values, so that the behaviour of `INTL_CHECK_LOCALE_LEN()` can be changed later if needed.

Reply:
@cmb69 I am really sorry for pinging you.. But could you please have a look if this can be merged when you have some time? I believe the PR is ready to be merged because I fixed the stubs after this discussion in https://github.com/php/php-src/pull/4821/commits/8049acc9c3a6703e0e8080323ab7cdb0b5a94a3c.